### PR TITLE
Attempt to fix some web test flakiness.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
If the service is disposed when we make calls on it, we need to bubble up an exception that is understood by the callers.

This is another speculative fix for https://github.com/flutter/flutter/issues/149238.

### CODE REVIEW

**Review by mdebbar — APPROVED** [review]
RSLGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/vmservice.dart [hunk 1]
```diff
@@ -989,6 +989,9 @@ class FlutterVmService {
         }
       }
       return await extensionAdded.future;
+    } on vm_service.RPCError {
+      // Translate this exception into something the outer layer understands
+      throw VmServiceDisappearedException();
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/vmservice.dart:994]: Actually, as I recall, RPCError has some information, like a code? Should we preserve that information? Check for it?

**Review by christopherfujino — CHANGES_REQUESTED** [review]
Question

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**